### PR TITLE
Add functionality to change start states with error checking in Build mode

### PR DESCRIPTION
**Pull request by Necried**
This adds the above feature and deals with the issue mentioned in #105

Though one thing that could be a minor inconvenience - users can only currently change their machine type in Simulating mode, and that ties in with the behavior of modifying start states.

### CODE REVIEW

**Comment by CSchank** [comment]
Yes. There are two options for this:
1. Allow the user to choose DFA/NFA on both screens. 
2. Choose DFA/NFA from the start when a machine is created.

Option 2 is perhaps more consistent with how "most apps" work. You create a document of a certain type then you're stuck in that type. But that leads me to what I don't like about it: I think it will be very common to realize half way through designing a machine that "actually I wanted an NFA" or vice versa. So maybe option 1 is better? Now that DFA/NFA actually affects the Build mode in a more visceral way, it probably makes sense to have it in both places?

**Comment by Necried** [comment]
Yeah, in this context I rather like Option 1 as well; other than the flexibility of choice that the users have, they could also potentially learn better if they toggle between these two options, particularly in SImulating mode, to see what the differences between the two are.

**Comment by CSchank** [comment]
Yes. Okay. Would you be able to make this change as part of this PR, by any chance?

**Comment by Necried** [comment]
Yeah, I'll add it tonight, so don't approve this PR yet.

**Comment by CSchank** [comment]
Great, thank you!

**Comment by Necried** [comment]
@CSchank Should be good to merge now

**Comment by CSchank** [comment]
Great, thank you!! I wonder why Travis is so glitchy!

**Comment by Necried** [comment]
Travis just can't keep up after 1 year of inactivity...

File: src/SharedModel.elm [hunk 1]
```diff
@@ -19,3 +21,57 @@ init =
...
+                        white
+
+                     else
+                        darkGray
+                    )
+                |> move ( 0, -4 )
+            ]
+            |> move ( -winX / 2 + 52, winY / 2 - 32 )
+            |> notifyTap (changeMsg NFA)
+        ]
```
Comment (CSchank) [inline src/SharedModel.elm:77]: I like how this is refactored into its own function now!

**Review by CSchank — APPROVED** [review]
Looks good to me!

**Pull request merged**